src/workspacer.Shared/Window/IWindow.cs

```diff
@@ -1,15 +1,18 @@
 ﻿using System;
 using System.Drawing;
-using System.Collections.Generic;
-using System.Diagnostics;
-using System.Linq;
-using System.Text;
-using System.Threading.Tasks;
 
 namespace workspacer
 {
+    public delegate void WindowClosedDelegate();
+    public delegate void WindowUpdatedDelegate();
+    public delegate void WindowFocusedDelegate();
```

Comment:
I believe it's customary/conventional for events like this to have a first argument `object sender` (or in this case `IWindows sender`), to make it easier for the consumer of the event to use and correlate the event-data internally.